flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1313,13 +1274,26 @@ struct EmboxCommonConversion : public FIROpConversion<OP> {
       auto ty = boxEleTy.cast<fir::CplxType>();
       return doComplex(getKindMap().getRealBitsize(ty.getFKind()));
     }
-    if (auto ty = boxEleTy.dyn_cast<fir::CharacterType>()) {
-      TODO();
-    }
-    if (auto ty = boxEleTy.dyn_cast<fir::LogicalType>())
+    if (auto ty = boxEleTy.dyn_cast<fir::CharacterType>())
+      return doCharacter(getKindMap().getCharacterBitsize(ty.getFKind()),
+                         ty.getLen());
+    if (auto ty = boxEleTy.dyn_cast<fir::LogicalType>()) {
+      // TODO: doesn't the runtime need to know these are LOGICAL? Pretend they
+      // are INTEGER for now.
```

Comment:
I would think so, there is a `CFI_type_Bool` in `ISO_Fortran_binding.h`, that being said, that is probably not enough to cover all the logical types (I think it would work with LOGICAL(1) only, which not the default logical type)...

Reply:
I saw that one, but since we need at least 3 distinct variants I decided to punt.